Cod3rs_Growth/Coders_Growth.Web/wwwroot/services/Validacao.js

```diff
@@ -0,0 +1,98 @@
+sap.ui.define([
+    "sap/ui/core/library"
+], function(coreLibrary) {
+    "use strict";
+
+   
+
+    const stringVazia = "";
+    const ValueStateErro = coreLibrary.ValueState.Error;
+    const ValueStatePadrao = coreLibrary.ValueState.None;
+   
+    return {
+         i18n : null,
+         setI18Nmodel :function(_i18n){
+            this.i18n=_i18n;
+            },
+
+        validarMarca: function(inputMarca) {
+
+            if (inputMarca.getValue() == stringVazia) {
+                inputMarca.setValueState(ValueStateErro);
+                inputMarca.setValueStateText(this.i18n.getText("ValidacaoMarca"));
+                return false;
+            } else {
+                inputMarca.setValueState(ValueStatePadrao);
+                return true;
+            }
+        },
+
+        validarModelo: function(inputModelo) {
+
+            if (inputModelo.getValue() == stringVazia) {
+                inputModelo.setValueState(ValueStateErro);
+                inputModelo.setValueStateText(this.i18n.getText("ValidacaoModelo"));
+                return false;
+            } else {
+                inputModelo.setValueState(ValueStatePadrao);
+                return true;
+            }
+        },
+
+        validarCor: function(inputCor) {
+
+            if (inputCor.getValue() == stringVazia) {
+                inputCor.setValueState(ValueStateErro);
+                inputCor.setValueStateText(this.i18n.getText("ValidacaoCor"));
+                return false;
+            } else {
+                inputCor.setValueState(ValueStatePadrao);
+                return true;
+            }
+        },
+
+        validarMemoria: function(inputMemoria) {
+
+            if (inputMemoria.getValue() == stringVazia) {
+                inputMemoria.setValueState(ValueStateErro);
+                inputMemoria.setValueStateText(this.i18n.getText("ValidacaoMemoria"));
+                return false;
+            } else {
+                inputMemoria.setValueState(ValueStatePadrao);
+                return true;
+            }
+        },
+
+        validarData: function(inputAnoFabricado) {
+
+            if (inputAnoFabricado.getValue() == stringVazia) {
+                inputAnoFabricado.setValueState(ValueStateErro);
+                inputAnoFabricado.setValueStateText(this.i18n.getText("ValidacaoAnoFabricacao"));
```

Comment:
Ao invés de sempre chamar o i18n assim, vc pode criar um método que faça isso e usar ele. Facilita pra vc escrever e a legibilidade de quem for ver esse código depois